### PR TITLE
feat: add request method functionality to lnd and commando

### PR DESCRIPTION
**Pull request by bumi**
This implements the requestMethod for the lnd and commando connector exposing certain functions to webln.
This allows web apps to do much more with lighthing than payments.

issue: #1725

### CODE REVIEW

**Comment by kiwiidb** [comment]
We still have this issue of the error not being propageted back to the website eg. if `experimental-dual-fund` is not enabled.

**Comment by reneaaron** [comment]
> We still have this issue of the error not being propagated back to the website eg. if `experimental-dual-fund` is not enabled.

Is that something outside of our control? Can we catch those errors somehow?

**Comment by reneaaron** [comment]
Simplified the permission dialog for a bit: 
 - Remove "host" (has been in this screen 3 times, now 1 time)
 - Unified "Remember my choice and don't ask again"
 - Minor style fixes

image

**Review by reneaaron — APPROVED** [review]
Just tested everything with LND and worked out fine so far👌 

 - added translations for permissions for commando as well as LND
 - minor adjustments to permission screen

File: src/extension/background-script/connectors/commando.ts [hunk 1]
```diff
@@ -107,6 +139,32 @@ export default class Commando implements Connector {
     await this.ln.disconnect();
   }
 
+  get supportedMethods() {
+    return supportedMethods;
+  }
+
+  methodDescription(method: string) {
+    return `commando.${method}`;
```
Comment (bumi) [inline src/extension/background-script/connectors/commando.ts:147]: as we now have the description in the translation file and only use the method here I am wondering if we still need this method or if we move this convention to the `request` action. 

we can set the `description` there to ``${connector.constructor.name.toLowerCase()}.${method}`` 
then we don't need the `methodDescription` function in the connector. 

we only need to check if there is a translation for that - though as long as we have a whitelist we can make it a requirement to have a translation. 
it only will no longer work if we allow everything (and not only whitelisted ones) - though I'd for now still stick with this whitelist.

Comment (reneaaron) [reply src/extension/background-script/connectors/commando.ts:147]: Agree. Will refactor this part 👍

**Pull request merged**